Etabs_Adapter/CRUD/Read/Results/Result.cs

```diff
@@ -187,14 +189,12 @@ private void ReadResultsError(Type resultType)
                 List<string> idsOut = new List<string>();
                 foreach (object o in ids)
                 {
-                    object idObj;
-
                     if (o is string)
                         idsOut.Add((string)o);
                     else if (o is int || o is double)
                         idsOut.Add(o.ToString());
-                    else if (o is T && ((T)o).CustomData.TryGetValue(AdapterIdName, out idObj))
-                        idsOut.Add(idObj.ToString());
+                    else if (o is T && ((T)o).AdapterId(typeof(ETABSId)) != null)
+                        idsOut.Add(GetAdapterId<string>((T)o));
```

Comment:
just small efficiency thing, but could chage to something like the following to avoid have to search for the Id multiple times:
```suggestion
                    else if (o is T)
                    {
                         string id = GetAdapterId<string>((T)o);
                         if(id != null)
                             idsOut.Add(id);
                     } 
```

(WROTE THIS ON GITHUB SO HAVE NOT FULLY CHECKED IT WORKS)